ekm/slashing_protection_storage.go

```diff
@@ -0,0 +1,185 @@
+package ekm
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/bloxapp/eth2-key-manager/core"
+	ssz "github.com/ferranbt/fastssz"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bloxapp/ssv/logging"
+	"github.com/bloxapp/ssv/storage/basedb"
+)
+
+const (
+	highestAttPrefix      = prefix + "highest_att-"
+	highestProposalPrefix = prefix + "highest_prop-"
+
+	// versionPrefix is the version prefix key of slashing protection DB
+	versionPrefix = "version"
+	// GenesisVersion is the genesis version of the slashing protection DB
+	GenesisVersion = "0x0"
+)
+
+type SpStorage interface {
```

Comment:
usually abbreviations are written with all caps, like DB, so you could do SPStorage